src/Libraries/Microsoft.Extensions.AI.Abstractions/Utilities/AIJsonUtilities.Defaults.cs

```diff
@@ -42,30 +42,18 @@ public static partial class AIJsonUtilities
     [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2026", Justification = "DefaultJsonTypeInfoResolver is only used when reflection-based serialization is enabled")]
     private static JsonSerializerOptions CreateDefaultOptions()
     {
-        // If reflection-based serialization is enabled by default, use it, as it's the most permissive in terms of what it can serialize,
-        // and we want to be flexible in terms of what can be put into the various collections in the object model.
-        // Otherwise, use the source-generated options to enable trimming and Native AOT.
-        JsonSerializerOptions options;
+        // Copy configuration from the source generated context.
```

Comment:
This is not fixing any bugs, just simplifying and removing a bit of duplication in configuration. It also prefers source generated models for known types even in the case where reflection is enabled, which should theoretically result in better startup times.